docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+= Git Submodule pattern
+
+== How this workflow pattern works:
+
+1. Create a "downstream" repo.
+2. Track your upstream git repo with a git submodule reference.
+3. Onboard to {ProductName}, use link:https://github.com/renovatebot/renovate[Renovate] to keep your Git submodule references up to date.
+4. In your downstream repo, apply patches in your Containerfile if you need to carry downstream patches.
+
+== Implementing this solution pattern
+=== Create a downstream repo
+
+A **downstream repo** is a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+The downstream repo will:
+
+* be used to hold the git submodule reference to the upstream repository.
+* contain a Containerfile that references and pulls in the git submodule(s) content.
```

Comment:
This isn't _strictly_ a requirement.

> and pulls in the git submodule(s) content

What do you mean by this?

Reply:
Replaced "will" with "can" to emphasize it's not a requirement.

For "and pulls in the git submodule(s) content" I was referencing the [git-clone build definition](https://github.com/konflux-ci/build-definitions/blob/main/task/git-clone/0.1/git-clone.yaml#L223-L231) that @ralphbean mentioned when answering my question (this is in the slack thread I added you to):
> In the Git submodule example, I see the Containerfile referencing the oras git submodule directory:
ENV ORASPKG /oras
>
>Is there some external service that pulls in that git submodule when the Containerfile is being built?
>
>For example, if I pull in this repo locally: https://github.com/konflux-ci/oras-container
and I go to the oras directory, it's empty until I run the following:
git submodule update --init --recursive
>
>I'm wondering at what point does this repo get pulled in so the Containerfile can use it 

The "Git submodule example" being this one: https://github.com/konflux-ci/oras-container

Reply:
I think the word "pull" was confusing me. The Containerfile will `COPY` the submodule to the stage so that the content can be built/leveraged for the resulting container.